include/library/spdm_common_lib.h

```diff
@@ -447,6 +447,49 @@ void spdm_register_transport_layer_func(
 	IN spdm_transport_encode_message_func transport_encode_message,
 	IN spdm_transport_decode_message_func transport_decode_message);
 
+/**
+  Verify a SPDM cert chain in a slot.
+
+  This function shall verify:
+    1) The integrity of the certificate chain. (Root Cert Hash->Root Cert->Cert Chain)
+    2) The trust anchor. (Root Cert Hash/Root cert matches the trust anchor)
+
+  The function shall check the negotiated hash algorithm to check root cert hash.
+  The function shall check the negotiated (req) asym algorithm to determine if it is right cert chain.
+
+  The function returns error if either of above is not satisfied.
+
+  @param  spdm_context                  A pointer to the SPDM context.
+  @param  slot_id                       The number of slot for the certificate chain.
+  @param  cert_chain_size               Indicate the size in bytes of the certificate chain.
+  @param  cert_chain                    A pointer to the buffer storing the certificate chain

```

Comment:
How should this behave if the Responder does not return the root certificate? Would the caller have to append the root certificate to the beginning of the certificate chain?

Reply:
This cert_chain contains everything returned from responder, including root_cert hash.

The verifier can determine which root_cert should be used to verify the chain, by comparing root_cert hash.

libspdm does not the trust anchor. Only verifier knows the trust anchor.

Reply:
Ok. Understood.